escrow/contracts/Escrow.cdc

```diff
@@ -0,0 +1,192 @@
+            }
+
+            // Create the entry and add it to the entries map
+            let entry <- create LeaderboardEntry(
+                nftID: nftID,
+                ownerAddress: depositCap.address,
+                nft: <-nft,
+                depositCap: depositCap
+            )
+
+            self.entries[nftID] <-! entry
+
+            emit NFTDeposited(leaderboardName: leaderboardName, nftID: nftID, owner: depositCap.address)
+        }
+
+        pub fun getEntriesLength(): Int {
+            return self.entries.keys.length
+        }
+
+        access(contract) fun withdraw(nftID: UInt64) {
+            let entry <- self.entries.remove(key: nftID)!
+            entry.withdraw()
+            emit NFTWithdrawn(leaderboardName: self.name, nftID: nftID, owner: entry.ownerAddress)
+            destroy entry
+        }
+
+        access(contract) fun burn(nftID: UInt64) {
+            let entry <- self.entries.remove(key: nftID)!
+            emit NFTBurned(leaderboardName: self.name, nftID: nftID)
+            destroy entry
+        }
+
+        // Destructor for Leaderboard resource.
+        destroy() {
+            destroy self.entries
+        }
+
+        init(name: String, nftType: Type) {
+            self.name = name
+            self.nftType = nftType
+            self.entries <- {}
+        }
+    }
+
+    // The resource representing an NFT entry in a leaderboard.
+    pub resource LeaderboardEntry {
+        pub let nftID: UInt64
+        pub let ownerAddress: Address
+        pub let nft: @{UInt64: NonFungibleToken.NFT}
+        pub let depositCapability: Capability<&{NonFungibleToken.CollectionPublic}>
+
+        pub fun withdraw() {
+            if self.depositCapability.check() {
+                let receiver = self.depositCapability.borrow()
+                    as &{NonFungibleToken.CollectionPublic}?
+                    ?? panic("Could not borrow the NFT receiver from the capability")
+
+                let nft <- self.nft.remove(key: self.nftID)!
+                receiver!.deposit(token: <- nft)
+            } else {
+                panic("Deposit capability is not valid")
+            }
+        }
+
+        // Destroys the NFT.
+        destroy() {
+            destroy self.nft
+        }
+
+        init(nftID: UInt64, ownerAddress: Address, nft: @NonFungibleToken.NFT, depositCap: Capability<&{NonFungibleToken.CollectionPublic}>) {
+            self.nftID = nftID
+            self.ownerAddress = ownerAddress
+            self.nft <- {nftID: <-nft}
+            self.depositCapability = depositCap
+        }
+    }
+
+    // An interface containing the Admin function that gets leaderboards by name.
+    pub resource interface IAdmin {
+        pub fun getLeaderboard(name: String): &Leaderboard?
+    }
+
+    // The resource representing an admin.
+    pub resource Admin: IAdmin {
+        // A dictionary holding leaderboards.
+        pub var leaderboards: @{String: Leaderboard}
```

Comment:
We should make this a contract level declaration with an `access(self)` and create structs to use to access the data. Like a `LeaderBoardData` struct that can be returned to anyone to access the leaderboard data

Reply:
Lets discuss this offline.